lib/global.neon

```diff
@@ -142,6 +142,13 @@ DECLARE EXCEPTION Utf8Encoding
  |%
 DECLARE EXCEPTION ValueRange
 
+%|
+ |  Exception: PastEndOfString
+ |
+ |  An attempt was made to read a substring, with a length that goes beyond the end of the string.
```

Comment:
This description is too specific. The exception could be used for things than other than substrings. For consistency with the other (currently existing) exceptions, something like simply:

> A string index is not valid or out of range.